nixos/modules/testing/test-instrumentation.nix

```diff
@@ -55,7 +55,14 @@ with import ../../lib/qemu-flags.nix { inherit pkgs; };
     systemd.services."serial-getty@hvc0".enable = false;
 
     # Only use a serial console, no TTY.
-    virtualisation.qemu.consoles = [ qemuSerialDevice ];
+    virtualisation.qemu.consoles = [ "hvc1" qemuSerialDevice ];
+
+    # SSH+socket backdoor, see "Debugging NixOS tests" section in NixOS manual
+    services.openssh = {
+      enable = true;
+      permitRootLogin = "yes";
+    };
+    users.extraUsers.root.initialPassword = "backdoor";
```

Comment:
Do we really need to set a password?
```
      services.openssh.enable = true;
      services.openssh.permitRootLogin = "yes";
      services.openssh.extraConfig = "PermitEmptyPasswords yes";
      users.extraUsers.root.password = "";
```

Reply:
Not really. I was hesitant to do that, but then again once it has a backdoor, why even bother with passwords.

Reply:
Done.

Reply:
There is an empty password already set in this file: https://github.com/NixOS/nixpkgs/blob/922916c51e2d9f664fd80c659bdb21047e0fb464/nixos/modules/testing/test-instrumentation.nix#L136